src/daggy.js

```diff
@@ -69,11 +73,17 @@
   }
 
   function sum$cata(fs) {
-    var tag = this[TAG];
-    if (!fs[tag]) {
-      throw new TypeError("Constructors given to cata didn't include: " + tag);
+    var tags = this.constructor[TAGS];
+    var tag;
+    for (var idx = 0; idx < tags.length; idx += 1) {
+      tag = tags[idx];
```

Comment:
Let's replace the four lines above with the following:

```javascript
this.constructor[TAGS].forEach(function(tag) { ... });
```

Reply:
I made the change, but it is now significantly slower than with the for loop, which was already significantly slower than not checking other tags. Still a plenty fast operation, what do you think?

Reply:
I think we should use for as it's faster.

Reply:
Why do you prefer forEach @davidchambers ?

Reply:
I find the `Array#forEach` equivalent clearer, but let's stick with the `for` loop since it performs better.